lib/plugins/rotate_asg_instances/asg.rb

```diff
@@ -297,7 +301,11 @@ def shutdown_instance(id)
 
       def instance_in_terminal_state?(instance)
         state = if instance.is_a?(Aws::EC2::Instance)
-                  instance.state.name
+                  if instance.exists?
+                    instance.state.name
+                  else
+                    "Terminated"
+                  end
                 elsif instance.is_a?(Aws::AutoScaling::Instance)
                   instance.load.lifecycle_state
```

Comment:
I think there is a similar condition here where `instance.load` will fail because the instance no longer exists.  We should probably handle that case here pro-actively if we can determine from documentation or the SDK source what the response would be in that situation.

Reply:
Have checked the AWS SDK for the instance.load functionality and have few points here.
1. There are two instances that we are currently handling 
a. EC2::Instance ( **EC2I** ) 
b. AutoScaling::Instance ( **ASGI** )

2. The place where it uses EC2I, it handles the exception, in case load is failed.
3. The place where it uses ASGI, there as well, it checks if data exists or not.
4. For the last one in line#310, I think, I can check for nil while checking for the terminal state for ASGI.